playwright/tests/e2e/steps/transactions-overview.steps.ts

```diff
@@ -0,0 +1,221 @@
+import { createBdd } from "playwright-bdd";
+import { expect } from "@playwright/test";
+
+import { koiosApi } from "playwright/api/call-koios/koios.api";
+
+import { transactionOverviewPage } from "../../../pages/transaction-overview.page";
+import { transactionDetailPage } from "../../../pages/transaction-detail.page";
+
+const { Given, When, Then } = createBdd();
+
+let addressId: string | null;
+let transactionHash: string | null;
+
+Given(/^the user is in the general dashboard page in explorer portal test$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToDashboard();
+});
+When(/^the user selects the Transactions option inside the Blockchain drop down menu$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToEpochsFromSidebar();
+});
+Then(
+  /^the user should see the transactions page containing the Search bar and Transactions table$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).searchBarOnTransaction();
+    await transactionOverviewPage(page).checkTransactionsTable();
+  }
+);
+
+Given(/^the user is in the transactions page in explorer portal site$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects any data of the row execpt the transaction hash, block number, epoch number or addresses hash of one of the transactions record in the table$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).openWidget();
+  }
+);
+Then(
+  /^the user should see the Info widget data with the respective data of the selected transaction$/,
+  async ({ page, request }) => {
+    const trxHash = await page.getByTestId("transaction.table.value.txhash#0");
+    const ariaLabelValue = await trxHash?.getAttribute("aria-label");
+    if (ariaLabelValue) {
+      (await koiosApi(request).getTransactionByTrxHash(ariaLabelValue)).json().then(async (data) => {
+        await transactionOverviewPage(page).checkCurrenTransactionWidget({ currentTransaction: data });
+      });
+    }
+  }
+);
+
+Given(/^the user is in the Transactions page$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the transaction hash of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transaction.table.value.txhash#0");
```

Comment:
Once again is better keep the logic of locators interactions in the respective pages, this code should be encapsulated in a method inside transactionsOverviewPage like open transaction details page by Tx hash or something like that and then you just call that method from here, if you need the value of the TxHash that method should return it.